api/include/mfxadapter.h

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) 2019 Intel Corporation
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+#if (MFX_VERSION >= MFX_VERSION_NEXT)
+#ifndef __MFXADAPTER_H__
+#define __MFXADAPTER_H__
+
+#include "mfxstructures.h"
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+mfxStatus MFXQueryAdapters(mfxComponentInfo* input_info, mfxAdaptersInfo* adapters);
+mfxStatus MFXQueryAdaptersDecode(mfxBitstream* bitstream, mfxU32 codec_id, mfxAdaptersInfo* adapters);
+mfxStatus MFXQueryAdaptersNumber(mfxU32* num_adapters);
```

Comment:
Per my opinion this API is over complex. For example, 2 structs with almost the same name (mfxAdapterSInfo) confuse. I would this:
```
mfxU32 MFXQueryAdaptersNumber();

// num_adapters size of the adapters array allocated by the user
// functions return number of match adapters within adapters array
mfxU32 MFXQueryAdapters(mfxComponentInfo* input_info, mfxAdapterInfo* adapters, mfxU32 num_adapters);
mfxU32 MFXQueryAdaptersDecode(mfxBitstream* bitstream, mfxU32 codec_id, mfxAdapterInfo* adapters, mfxU32 num_adapters);
```

Reply:
But we also report back internal statuses, which is aligned with MSDK API, where we always return statuses

Reply:
This is not an argument to me. Imho, "we always did that" is not kind of argument I would accept. I still am persistent with my opinion: API is over complex.

Reply:
Well, from other side your approach hiding errors. What if something will go wrong? You suggest to return 0 in such case (no suitable adapters), but how user will know the reason of this? Either it is workload that can't be handled, or some internal error occured